doc/error_handling_guidelines.md

```diff
@@ -0,0 +1,130 @@
+# Azure CLI Error Handling Guidelines
+
+This document aims to provide the guidelines for command group authors to onboard the error handling method and the new error output format.
+
+Previously in Azure CLI, `CLIError` is widely used to wrap the error messages after the different kinds of error types being caught. Now, `CLIError` is deprecated and it is replaced by a set of new designed error types, which will provide more clear structures for error categorizing and also more actionable error outputs for user experience.
+
+For the new commands authoring, here are what need to be done to get on board. __Any PRs not following rules here will be rejected.__ For the existing `CLIError` in command groups, command group owners should schedule to replace them with the new error types.
```

Comment:
How about extensions? They need to update the minimum version in metadata if they want to use the new error type, right?

Reply:
Theoretically yes. 

But if we force the extensions to use the new error types, it will make the extensions not compitable to the old versions of AzureCLI. This may need more discussions